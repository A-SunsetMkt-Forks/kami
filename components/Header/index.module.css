.header {
  @apply fixed top-0 left-0 right-0;
}

.head-swiper {
  @apply relative flex items-center;

  height: 4rem;
}

@screen phone {
  .head-info {
    display: none;
  }
}

.title {
  @apply relative inline-block text-xl font-medium mt-0 ml-4 truncate flex-shrink-0;
}

.more-button {
  @apply invisible hidden;
}

@screen phone {
  .more-button {
    @apply absolute right-0 visible block cursor-pointer py-4 pl-4;
  }
}

.nav-container {
  height: 4rem;

  @apply mx-20 relative overflow-hidden;
}

.header-logo svg {
  height: 2rem;
}

@screen phone {
  .header nav {
    @apply mx-4;
  }

  .title {
    @apply ml-1;
  }
}
.link-group {
  @apply relative inline-flex items-center;

  height: 4rem;

  & .anchor-ball {
    @apply absolute left-0;
  }
}

@screen phone {
  .link-group {
    @apply hidden;
  }
}

.link-item {
  @apply inline-flex items-center px-4 cursor-pointer;

  height: 4rem;
}

.link-title {
  @apply ml-3;
}

.drawer {
  @apply fixed z-90 top-0 bottom-0 right-0 bg-light-bg overflow-auto;
  @apply transition-transform duration-300 p-6 text-light-brown;

  width: 300px;
  max-width: 80vw;
  transform: translateX(100%);

  &.show {
    transform: translateX(0);
  }

  & a {
    color: inherit;
  }

  & .parent,
  & .children {
    @apply py-2;
  }

  & .parent {
    @apply font-medium;
    @apply text-xl;
  }
  & .children-wrapper {
    @apply grid;

    grid-template-columns: 1fr 1fr;
  }
  svg {
    @apply mr-4;
  }
}

.link-section {
  @apply mb-4;
}
.nav-fake {
  @apply overflow-visible;
}
