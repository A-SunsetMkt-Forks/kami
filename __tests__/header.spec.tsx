/*
 * @Author: Innei
 * @Date: 2020-06-13 11:30:08
 * @LastEditTime: 2020-06-13 13:32:25
 * @LastEditors: Innei
 * @FilePath: /mx-web/__tests__/header.spec.tsx
 * @Coding with Love
 */
import React from 'react'
import { render, screen } from '@testing-library/react'
import Header from '../components/Header'
it('renders homepage unchanged', () => {})
