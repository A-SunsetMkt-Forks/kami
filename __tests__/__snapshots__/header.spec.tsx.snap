// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<header
  className=""
  style={
    Object {
      "backdropFilter": undefined,
      "backgroundColor": undefined,
    }
  }
>
  <style
    jsx={true}
  >
    
          .head-logo {
            display: flex;
            align-items: center;
          }
          header {
            background-color: var(--bg-opacity);
            backdrop-filter: blur(25px) saturate(150%) brightness(1.1);
            z-index: 5;
            transition: transform 0.5s, backdrop-filter 0.5s;
          }
          header.hide {
            transform: translateY(-100%);
          }
          @media screen and (max-width: 600px) {
            .head-logo {
              display: block;
            }
          }

          html.dark header {
            backdrop-filter: saturate(150%) blur(30px);
          }
        
  </style>
  <div
    className="head-logo"
    onClick={[Function]}
  >
    <svg
      version="1.1"
      viewBox="0 0 200 200"
    >
      <desc>
        Created with Sketch.
      </desc>
      <g
        fill="none"
        fillRule="evenodd"
        id="Custom-Preset"
        stroke="none"
        strokeWidth="1"
      >
        <g
          id="Group"
          transform="translate(1.000000, 0.000000)"
        >
          <g
            fillRule="nonzero"
            transform="translate(0.891670, 0.001664)"
          >
            <g
              fill="currentColor"
              id="Shape-2"
            >
              <path
                d="M97.9232558,0 C43.8413297,0 0,43.8413297 0,97.9232558 C0,152.005182 43.8413297,195.846512 97.9232558,195.846512 C152.005182,195.846512 195.846512,152.005182 195.846512,97.9232558 C195.846512,43.8413297 152.005182,0 97.9232558,0 Z M97.9232558,184.96615 C49.8516415,184.96615 10.8803618,145.99487 10.8803618,97.9232558 C10.8803618,49.8516415 49.8516415,10.8803618 97.9232558,10.8803618 C145.99487,10.8803618 184.96615,49.8516415 184.96615,97.9232558 C184.96615,145.99487 145.99487,184.96615 97.9232558,184.96615 Z"
                id="Shape"
              />
            </g>
            <circle
              cx="97.9232558"
              cy="97.9232558"
              id="Oval"
              r="78.8162791"
            />
          </g>
          <g
            id="Shape"
            stroke="currentColor"
            strokeWidth="8.8"
            transform="translate(48.391670, 47.501664)"
          >
            <path
              d="M50.1120234,100.116279 L0,50.1120234 L50.1120234,0 L100.116279,50.0042556 L50.1120234,100.116279 Z M90.5813953,57.2093023 L72.7145681,40.3711034 L50.1119553,19.0697674 L9.53488372,57.2093023 M71.5116279,42.9069767 L33.372093,81.0465116 M28.6046512,42.9069767 L66.744186,81.0465116"
            />
          </g>
        </g>
      </g>
    </svg>
  </div>
  <div
    className="head-swiper"
  >
    <nav
      className="head-info"
    >
      <div
        className="desc"
      >
        <div
          className="meta"
        />
        <div
          className="title"
        />
      </div>
    </nav>
    <nav
      className="head-menu head-menu"
    />
  </div>
  <nav
    className="head-social"
  />
</header>
`;
